ui/src/chatbot.tsx

```diff
@@ -115,6 +164,7 @@ export const XChatbot = (props: Chatbot) => {
     botTextColor = React.useMemo(() => getContrast(theme.palette.neutralLighter), [theme.palette.neutralLighter]),
     msgContainerRef = React.useRef<HTMLDivElement>(null),
     skipNextBottomScroll = React.useRef(false),
+    hasSomeCaption = React.useMemo(() => props.prompt_suggestions?.some(({ caption }) => !!caption), [props.prompt_suggestions]),
```

Comment:
No need for memo here. We expect a few suggestions only so safe to recompute on every render.